tensorboard/webapp/widgets/data_table/content_cell_component.ts

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
+import {ColumnHeader, ColumnHeaderType} from './types';
+import {
+  intlNumberFormatter,
+  numberFormatter,
+  relativeTimeFormatter,
+} from '../line_chart_v2/lib/formatter';
+
+@Component({
+  selector: 'tb-data-table-content-cell',
+  template: `
+    <ng-container [ngSwitch]="header.type">
+      <div *ngSwitchCase="ColumnHeaderType.VALUE_CHANGE" class="cell">
+        <ng-container
+          *ngTemplateOutlet="arrow; context: {$implicit: datum}"
+        ></ng-container>
+        {{ getFormattedDataForColumn() }}
+      </div>
+      <div *ngSwitchCase="ColumnHeaderType.PERCENTAGE_CHANGE" class="cell">
+        <ng-container
+          *ngTemplateOutlet="arrow; context: {$implicit: datum}"
+        ></ng-container>
+        {{ getFormattedDataForColumn() }}
+      </div>
+      <div *ngSwitchDefault class="cell extra-right-padding">
+        {{ getFormattedDataForColumn() }}
+      </div>
+      <ng-content></ng-content>
+    </ng-container>
+    <ng-template #arrow let-value>
+      <mat-icon *ngIf="value >= 0" svgIcon="arrow_upward_24px"></mat-icon>
+      <mat-icon *ngIf="value < 0" svgIcon="arrow_downward_24px"></mat-icon>
+    </ng-template>
+  `,
+  styles: [
+    `
+      :host {
+        display: table-cell;
+        padding: 1px;
+      }
+      .cell {
+        align-items: center;
+        display: flex;
+      }
+
+      .cell mat-icon {
+        height: 12px;
+        width: 12px;
+
+        ::ng-deep path {
+          fill: unset;
+        }
+      }
+
+      .extra-right-padding {
+        padding-right: 1px;
+      }
+    `,
+  ],
```

Comment:
At this size I would like to see the template and styles broken out into their own components

Reply:
Done.